src/Collection.php

```diff
@@ -1112,6 +1131,13 @@ public function withOptions(array $options = [])
         return new Collection($this->manager, $this->databaseName, $this->collectionName, $options);
     }
 
+    private function inheritBuilderEncoder(array $options): array
+    {
+        $options['builderEncoder'] = $this->builderEncoder;
+
+        return $options;
```

Comment:
You could make this a little more concise:
```suggestion
        return $options + ['builderEncoder' => $this->builderEncoder];
```

I'm aware this would allow passing a `builderEncoder` option to any operation that needs to handle encoding itself, so feel free to ignore if you'd rather not do this.

Reply:
> I'm aware this would allow passing a `builderEncoder` option to any operation that needs to handle encoding itself, so feel free to ignore if you'd rather not do this.

That's a feature we don't want to support now.

The equivalent code to what I've done would therefore be:
```suggestion
        return ['builderEncoder' => $this->builderEncoder] + $options;
```